### PR TITLE
fix(nfl): league-aware summary, radar scale, colour ramp and logos on the preview NFL pages

### PR DESCRIPTION
**Pull request by saiemgilani**
## Summary
Bug fixes to the NFL pages, from the preview triage of `main` (4ee954df). All of
`pages/nfl/**` is behind `nfl: 'preview'` in `astro/src/utils/features.ts`, so every
surface touched here is preview-only: **no new flag, nothing public changes**, and no
`classic/` twin is involved (these are the pregame/team/season components, not the
game page).

Findings implemented (numbers from the triage report):

- **#3 — NFL pregame pages lost venue, weather, betting line and Recent Form.**
  `retrieveGameSummary()` hardcoded the `college-football` summary path, a 404 for every
  NFL event. It now takes a `league` and builds the path from the existing
  `LEAGUES[league].espnPath`; `PreGamePage.astro` is the only caller.
- **#4 — every NFL radar was a near-full circle.** `generatePercentile(rank, 134)` on all
  16 axes. The league is threaded into the radar builders from the two chart islands
  (each takes a `league` prop, with `leagueFromLocation()` as the `client:only` fallback)
  and the scale is `LEAGUES[league].teamCount`.
- **#5 — every NFL rank cell was dark green.** `generateColorRampValue(rank, 134, true)`
  in `TeamCard.astro` (5 cells) and `MatchupView.astro` → `cfg.teamCount`, the pattern
  `TeamProfileRoute.astro:45` already used.
- **#6 — NFL pregame pages showed college logos** (Akron's logo for the Atlanta Falcons).
  `teamlogos/ncaa/500/` → `espnLogoLeague(Astro.locals.league)` in `PlayersToWatch`,
  `RecentForm` and `GameEssentials`. The Players-to-Watch team link was also unprefixed
  (`/team/<id>` on an NFL page); it now goes through `leaguePath`.
- **#7 — college copy on NFL pages.** "FBS vs FBS games only" (`TeamRadarChart.svelte`)
  and the collegefootballdata.com credit (`SeasonTeamRoute.astro`) render only for `cfb`.
- **Reader switch for #1/#2.** The data side lands in sdv-db as `nfl.team_info` +
  `nfl.espn_schedule`, both cfb-shaped. `team_info` keeps its name; the schedule table is
  now named in the league config (`scheduleTable`: `schedule` for cfb, `espn_schedule` for
  nfl) and read from there in `retrieveTeamGames`, which is what the team record, the
  season schedule and the matchup history all go through. The NFL `schedule` table is the
  nflverse frame and carries none of the columns the site filters/orders on.

Not in this PR: finding #1's `retrieveTeam` mapping (data-side fix) and #8 (the Schedule
panel is commented out on both leagues, pre-dating #229).

**Collision check:** `gh api repos/saiemgilani/game-on-paper-app/pulls/270/files` lists
none of the files here. The one adjacency is `MatchupHistoryTable.astro` (in #270), which
sits inside `MatchupView` — it is untouched here. `utils/misc.ts` is untouched too (#269
and #270 both hold it); the colour-ramp and logo fixes are new *arguments at call sites*,
not new helpers.

## Tests
`cd astro && npx vitest run` → **43 files, 352 passed, 1 skipped** (was 42/342+1).

New `astro/test/preGamePage.render.test.ts` renders `PreGamePage` for both leagues from
two real captured ESPN pregame payloads (`pregame-401872933-nfl.json.gz` CAR @ ATL,
`pregame-401858458-cfb.json.gz` PUR @ UCLA; provenance in `test/fixtures/README.md`), with
the SDV client mocked. The existing NFL game-page fixture is a *final* game, so
`PreGamePage` never rendered in the suite and its `not.toContain('teamlogos/ncaa/')`
assertion passed vacuously — that is the gap this closes. It asserts per league: the
summary endpoint actually requested, NFL logos with no `ncaa/500/` path, the 32-team
colour ramp (`hulk-bg-level-3` for a 24th-place rank, vs `-8` on the cfb twin), the
`/nfl`-prefixed team links, the `espn_schedule` vs `schedule` table split, and the radar
percentile scale (24th of 32 = 25, 24th of 134 = 82, worst-in-league = 0).

`python/` untouched. `npx astro check` reports no new errors in the changed files (the
278 pre-existing errors are the `Locals.cfContext` class every render test already trips).
No local four-shot matrix: this host can't run the preview Worker, so the evidence
workflow's screenshots are the visual check.

## Evidence
Evidence routes: /nfl/game/401872980 /nfl/team/7 /team/130

`/nfl/game/401872980` is TB @ DAL on 2026-10-09, scheduled for weeks yet, so the pregame
page stays a pregame page across re-runs. `/team/130` is the cfb twin of the NFL team
page, since the two share every component changed here. The workflow forces `preview`
flags on in both trees, so `/nfl` renders without a preview cookie.

## Checklist
- [x] vitest + pytest pass locally (pytest N/A: no python change)
- [x] `Evidence routes:` names the pages this change affects most
- [ ] The pr-evidence comment is green, or every regression it flags is explained here
- [x] Preview-gated work stays behind its flag in `astro/src/utils/features.ts`

## Summary by Sourcery

Make preview NFL pages league-aware across pregame data, schedules, visualisations, branding, and matchup history.

New Features:
- Display all completed meetings for a matchup, including multiple NFL divisional meetings, with league-aware labels and links.

Bug Fixes:
- Restore NFL pregame venue, weather, betting, and recent-form data by requesting league-specific ESPN summaries.
- Use league-specific schedule data, logos, team links, rank colour scales, radar percentiles, and copy across NFL and college football pages.
- Exclude scheduled games from matchup history and correctly identify NFL postseason rounds across different season formats.

Enhancements:
- Thread league context through radar charts and matchup components so visualisations and team presentation use the correct league configuration.

Documentation:
- Document the new scheduled pregame fixtures and their provenance for rendering tests.

Tests:
- Add league-aware pregame rendering coverage for ESPN endpoints, logos, links, schedules, colour ramps, and radar scales.
- Add matchup rendering tests covering zero, single, and multiple completed meetings.
- Add unit coverage for NFL week labels and matchup labels.

## Data side

Findings 1-2 (Unknown Team, 0-0 record / empty schedule) are closed by sportsdataverse/sdv-db#73 — `nfl.team_info` and `nfl.espn_schedule` (cfb.schedule-shaped, keyed by ESPN ids) are already live on the Data API, so `scheduleTable` here reads real rows. `conference_game` there is nflverse `div_game`, hence the `Div:` label (da0d4d2e).

## Summary by CodeRabbit

- **New Features**
  - Added league-aware support across matchup, team, radar, and pre-game views, including NFL-specific links, logos, schedules, and rank scaling.
  - Radar charts now use the correct team-count scale for each league.
  - Matchups can display multiple completed meetings from the same season with game-by-game links and labels.
  - Added NFL week and postseason round labels.

- **Bug Fixes**
  - Corrected team logos, navigation paths, game summaries, and schedule data for non-college leagues.
  - Limited matchup history to completed games.

### CODE REVIEW

File: astro/src/components/game/MatchupView.astro [hunk 1]
```diff
@@ -4,6 +4,7 @@ import { formatYardline, roundNumber, generateColorRampValue, generateMarginalSt
 import MatchupRadarChart from '../charts/MatchupRadarChart.svelte';
 import FallbackSpinner from '../FallbackSpinner.astro';
 import MatchupHistoryTable from './MatchupHistoryTable.astro';
+import { DEFAULT_LEAGUE, LEAGUES } from '../../utils/league';
```
Comment (akeaswaran) [inline astro/src/components/game/MatchupView.astro:7]: Matchup history / Previous matchups should not include unplayed games (See: CAR/ATL - 2026 Week 3)

File: astro/src/components/game/MatchupView.astro [hunk 2]
```diff
@@ -4,6 +4,7 @@ import { formatYardline, roundNumber, generateColorRampValue, generateMarginalSt
...
 
 interface Props {
```
Comment (akeaswaran) [inline astro/src/components/game/MatchupView.astro:9]: Additional note: MatchupBuilder will look for a previous matchup given two teamSeasons that share a season value and show a link to that game if found. The problem is that NFL division opponents play at least twice in a given year. We should handle this case better

File: astro/src/components/game/MatchupView.astro (hunk 1 above)
Comment (saiemgilani) [reply astro/src/components/game/MatchupView.astro:7]: Fixed in 2eccdd07. Root cause was the reader, not the table component: `nfl.espn_schedule` (and `cfb.schedule`) hold the whole season, and `retrieveMatchupHistory` in `resources/sdv.ts` passed every row straight through, so CAR/ATL 2026 (both the September game and the week 18 one) rendered under Previous Meetings with no score.

`retrieveMatchupHistory` now keeps only rows with `completed === true` and both point totals present. The filter is there rather than in `retrieveTeamGames`, which the team-season schedule panels read and which legitimately wants future games — one place, and both callers (PreGamePage and MatchupRoute) get it.

`MatchupHistoryTable.astro` untouched (#270 owns it). Covered in `test/preGamePage.render.test.ts` with the real ATL/CAR rows — two completed 2025 meetings and the scheduled 2027-01-10 one; the test fails without the filter (the unplayed row renders).

File: astro/src/components/game/MatchupView.astro (hunk 2 above)
Comment (saiemgilani) [reply astro/src/components/game/MatchupView.astro:9]: Handled in f44eb158. Two bugs, actually: the query was `home_id=<home>, away_id=<away>, limit: 1`, so it found at most one game and only if that side hosted — the away leg was unreachable.

It now calls `retrieveMatchupHistory(home, away, 5, league, season)` (same reader, newly season-scopable), which queries both orientations and returns every completed meeting oldest-first. One meeting keeps today's single-link alert; more than one renders a short list of links, e.g. `Week 3: ATL 0-30 CAR` / `Week 11: CAR 30-27 ATL`.

Round labels come from `weekLabel()` in `utils/league.ts`: `nfl.espn_schedule` uses nflverse numbering where the postseason continues the count, so weeks 19-22 read as Wild Card / Divisional / Conference Championship / Super Bowl instead of "Week 22"; cfb and anything unrecognised fall back to "Postseason". Copy follows docs/design-conventions.md — colon separator rather than a dot, abbreviation over full name.

Tests: `test/league.test.ts` for the labels, `test/matchupBuilder.render.test.ts` for the one/many/none branches. Full vitest: 44 files, 360 passed, 1 skipped. No visual check from here (`astro dev`/`build` need workerd, which will not run on this box) — the pr-evidence workflow shoots the matrix on push.

File: astro/src/utils/league.ts [hunk 3]
```diff
@@ -103,6 +103,38 @@ export function teamLogoUrl(league: League | undefined, teamId: string | number,
     return `https://a.espncdn.com/i/teamlogos/${logoLeague}/500-dark/${id}.png`;
 }
 
+// `nfl.espn_schedule` carries nflverse week numbering, where the postseason
+// CONTINUES the regular-season count instead of restarting at 1 -- so a game in
+// week 22 is the Super Bowl, and "Week 22" would mean nothing to a reader.
+const NFL_POSTSEASON_ROUNDS: Record<number, string> = {
```
Comment (akeaswaran) [inline astro/src/utils/league.ts]: These may be inaccurate for previous non-18 week seasons

Comment (saiemgilani) [reply astro/src/utils/league.ts]: Right — fixed in 28db6cc7. nflverse week numbers continue through the postseason, and the count they continue from moved: the regular season was 17 weeks through 2020 and 18 from 2021, so week 19 is the Divisional round in 2019 and the Wild Card round in 2021. `weekLabel` now takes the season and reads the round as `week - nflRegularSeasonWeeks(season)`. Checked against the DB: `nfl.schedule` and `nfl.espn_schedule` show exactly two patterns back to 1999 (1999-2020 REG 1-17 / WC-SB 18-21, 2021-2025 REG 1-18 / WC-SB 19-22) with no exceptions, so it is a season cutoff rather than a per-season lookup. Unit tests cover both eras, the 2021 boundary and an unknown season (reads as the current era).

**Comment by akeaswaran** [comment]
Agree with the coderabbit feedback, the links/params need to be updated

**Comment by saiemgilani** [comment]
Both CodeRabbit outside-diff findings fixed in d90df6d0: RecentForm links go through `leaguePath` (no more `/game/<id>` from NFL pregame pages) and PlayersToWatch passes `Astro.locals.league` to all three summary reads. The NFL pregame render test now asserts the `/v1/nfl/passing` request, no `/v1/cfb/passing`, and no bare `/game/` hrefs; it fails on the previous head.

**Review by akeaswaran — APPROVED** [review]
